src/genome.py

```diff
@@ -94,7 +100,13 @@ class ListGenome(Genome):
 
     def __init__(self, n: int):
         """Create a new genome with length n."""
-        ...  # FIXME
+        self.lst = []
+        self.ID = []
+        for i in range(n):
+            self.lst.append("-")
+            self.ID.append(0)
+        self.active_lst = []
```

Comment:
Is there any good reason to use a list here, rather than a set? Checking membership in a list takes $O(n)$ while checking in a list takes expected $O(1)$.